platform-plugins.md

```diff
@@ -0,0 +1,305 @@
+an API definition written in Dart, coupled with a platform-specific
+implementation for Android, for iOS, or for both.
+
+There are two ways you can use plugins:
+
+* **Use existing platform plugins:** For frequently used APIs and SDKs, use
+ existing platform plugins, with no need to write any platform-specific code.
+ [See details below](#use).
+
+* **Create a platform plugin:** Write your own platform-specific code using
+ Flutter's Platform Channels. This can be used for private APIs, or for cases
+ where a suitable platform plugin is not yet available. [See details below](#create).
+
+And, if desired, you can share plugins with others:
+
+* **Publish a platform plugin:** Publish your own custom platform-specific as a
+ plugin. This enables other developers to use it by simply declaring a
+ dependency on the plugin. [See details below](#publish).
+
+## Use existing platform plugins {#use}
+
+### Core Flutter platform plugin
+
+The Flutter framework includes a core platform plugin for foundational
+platform-specific services, such as:
+
+* [Clipboard](https://docs.flutter.io/flutter/services/Clipboard-class.html):
+Utility methods for interacting with the system's clipboard.
+
+* [SystemSound](https://docs.flutter.io/flutter/services/SystemSound-class.html):
+Provides access to the library of short system specific sounds for common tasks.
+
+* [pathProvider](https://docs.flutter.io/flutter/services/PathProvider-class.html):
+Returns commonly used locations on the filesystem.
+
+See the [services library](https://docs.flutter.io/flutter/services/services-library.html)
+for full details.
+
+**Example: Using the Flutter platform plugin to read the contents of the Clipboard**
+
+The core Flutter platform plugin is available through the [services
+library](https://docs.flutter.io/flutter/services/services-library.html), as
+illustrated in this example:
+
+```dart
+import 'package:flutter/services.dart';
+
+class _MyHomePageState extends State<MyHomePage> {
+ String clipboardContents = 'Unknown';
+
+ getClipboardContents() async {
+   var clipboardData = await Clipboard.getData(Clipboard.kTextPlain);
+   if (clipboardData != null) {
+     setState(() {
+       clipboardContents = clipboardData.text;
+     });
+   }
+ }
+ ...
+}
+```
+
+### Platform plugins from the Flutter ecosystem
+
+{% include note.html content="Sharing of platform plugins is currently not
+supported, but Flutter is expected to support this in the future." %}
+
+## Create a platform plugin {#create}
+
+Flutter uses a flexible system that allows you to call platform-specific API
+whether available in Java code on Android, or in ObjectiveC or Swift code on
+iOS.
+
+Flutter's platform-specific API support does not rely on code generation, but
+rather on a flexible message passing style:
+
+* The Flutter portion of your app sends messages to it's *host*, the iOS or
+ Android portion of your app, over a platform channel.
```

Comment:
Here platform channel is spelled in lower-case. Even better. We should be consistent.